firedrake/interpolation.py

```diff
@@ -299,39 +333,15 @@ def interpolate(self, *function, output=None, transpose=False, default_missing_v
             # V can be the Function to interpolate into (e.g. see `Function.interpolate``).
             output = self.V
 
-        get_working_tape()
-        # Assembly annotation records the form before it is preprocessed. Hence, we preprocess
-        # the form beforehand to facilitate derivation of the tangent linear/adjoint models.
-        # For example,
-        # -> `interp = Action(Interpolate(v1, v0), f)` with `v1` and `v0` being respectively `Argument`
-        # and `Coargument`. Differentiating `interp` is not currently supported as the action's left slot
-        # is a 2-form. However, after preprocessing, we obtain `Interpolate(f, v0)`, which can be differentiated.
-        interp = preprocess_base_form(interp)
-        res = assemble(interp, tensor=output, is_base_form_preprocessed=True)
+        # Assemble the `ufl.Interpolate` object, which will then call `Interpolator._interpolate`
+        # to perform the interpolation. Having this structure ensures consistency between
+        # `Interpolator` and `Interp`. This mechanism handles annotation since performing interpolation will drop an
+        # `AssembleBlock` on the tape.
+        res = assemble(interp, tensor=output)
         return res
 
     @abc.abstractmethod
     def _interpolate(self, *args, **kwargs):
```

Comment:
Please add a docstring to `_interpolate` so it's clear what it does (i.e. it performs the interpolation)